src/common/models/utils.ts

```diff
@@ -0,0 +1,54 @@
+import _ from 'lodash';
+
+// TODO(michael): We should perhaps create a Timeseries class that wraps an
+// Array<Date> and Array<number | null> and hang these utility functions off of
+// that.
```

Comment:
FWIW- I think this would actually make a lot of things cleaner.  We have a lot of utility functions for smoothing / manipulating timeseries data at this point...